python/ray/experimental/compiled_dag_ref.py

```diff
@@ -85,8 +92,17 @@ def __reduce__(self):
 
     def __del__(self):
         # If not yet, get the result and discard to avoid execution result leak.
-        if not self._ray_get_called:
-            self.get()
+        if not self._ray_get_called and not self._dag.is_teardown:
+            try:
+                self.get()
+            except Exception as e:
+                if RAY_IGNORE_UNHANDLED_ERRORS:
+                    return
+
+                logger.error(
+                    "Unhandled error (suppress with "
+                    f"{RAY_IGNORE_UNHANDLED_ERRORS_ENV_VAR}=1'): {e}"
+                )
```

Comment:
I think python by default ignores the exceptions happen in `__del__` and logs to stderr.

What's the difference with this change? Can you post the log before and after the change?

Reply:
that will print unnecessary stacktrace (to a destructor) and and an additional message the "exception raised in destructor is ignored". This will have clean error messages